ui-episodedetails/src/main/java/app/tivi/episodedetails/EpisodeDetails.kt

```diff
@@ -101,20 +102,47 @@ import com.google.accompanist.coil.rememberCoilPainter
 import com.google.accompanist.insets.navigationBarsHeight
 import com.google.accompanist.insets.navigationBarsPadding
 import com.google.accompanist.insets.statusBarsPadding
+import dagger.hilt.EntryPoint
+import dagger.hilt.InstallIn
+import dagger.hilt.android.EntryPointAccessors
+import dagger.hilt.android.components.ActivityComponent
 import org.threeten.bp.OffsetDateTime
 import kotlin.math.absoluteValue
 import kotlin.math.hypot
 
 @Composable
 fun EpisodeDetails(
+    id: Long,
     navigateUp: () -> Unit,
 ) {
+    // TODO: handle ContextWrappers and unwrap as necessary
+    val context = LocalContext.current as Activity
+
     EpisodeDetails(
-        viewModel = hiltViewModel(),
+        viewModel = remember(id) {
```

Comment:
Good usage of Hilt here 👍  I was wondering how we could propagate `onForgotten` calls to the `StateModel`. 

How safe would it be if apart from the `remember(id)` to create a `DisposableEffect(id)` that does that?

```
val viewModel = remember(id) { /* ... */ }
DisposableEffect(id) {
    onDispose { viewModel.onForgotten() }
}
```

We'll need to investigate implications when multi-threaded compositions take place. But for now... I think that should be safe to use?

Reply:
Yep. This is exactly how `rememberCoroutineScope()` is implemented. I also use this pattern in Accompanist.

Reply:
Ah, then `DisposableEffect` is not needed. TIL `RememberObserver` which already does all of that for you. That's great!

Reply:
It’s super powerful, I’ve been meaning to ping you and about doc-ing this pattern more.